pallets/pools/src/lib.rs

```diff
@@ -447,7 +447,7 @@ pub mod pallet {
 		/// Pool metadata was set.
 		MetadataSet {
 			pool_id: T::PoolId,
-			metadata: Vec<u8>,
+			metadata: BoundedVec<u8, T::MaxSizeMetadata>,
```

Comment:
A bit of a clean-up change, I can't think of any reason not to make this event param a bounded vec as well.